diff --git a/src/grandiose_receive.cc b/src/grandiose_receive.cc
index d8f6c94501b4221039d842d922cf903a6948a4b2..4f03b2f102d70d79df2129f5173eae9037b271a0 100644
--- a/src/grandiose_receive.cc
+++ b/src/grandiose_receive.cc
@@ -624,7 +624,7 @@ void audioReceiveComplete(napi_env env, napi_status asyncStatus, void* data) {
 }
 
 napi_value dataAndAudioReceive(napi_env env, napi_callback_info info,
-    char* resourceName, napi_async_execute_callback execute,
+    const char* resourceName, napi_async_execute_callback execute,
     napi_async_complete_callback complete) {
   napi_valuetype type;
   dataCarrier* c = new dataCarrier;
diff --git a/src/grandiose_util.cc b/src/grandiose_util.cc
index 833b6814bdfca144e0cb52b9ce9352fc741b729e..3dd4fcbe5a1df8f61943649ad7c3b91204a0addf 100644
--- a/src/grandiose_util.cc
+++ b/src/grandiose_util.cc
@@ -71,7 +71,7 @@ const char* getNapiTypeName(napi_valuetype t) {
   }
 }
 
-napi_status checkArgs(napi_env env, napi_callback_info info, char* methodName,
+napi_status checkArgs(napi_env env, napi_callback_info info, const char* methodName,
   napi_value* args, size_t argc, napi_valuetype* types) {
 
   napi_status status;
@@ -118,7 +118,7 @@ void tidyCarrier(napi_env env, carrier* c) {
   delete c;
 }
 
-int32_t rejectStatus(napi_env env, carrier* c, char* file, int32_t line) {
+int32_t rejectStatus(napi_env env, carrier* c, const char* file, int32_t line) {
   if (c->status != GRANDIOSE_SUCCESS) {
     napi_value errorValue, errorCode, errorMsg;
     napi_status status;
diff --git a/src/grandiose_util.h b/src/grandiose_util.h
index 07082e838c3c273a0e1e7d6d9fe3bff8b2c890a6..8e023badb93078988255552c68deb17811a0bd5d 100644
--- a/src/grandiose_util.h
+++ b/src/grandiose_util.h
@@ -50,7 +50,7 @@ napi_status checkStatus(napi_env env, napi_status status,
 long long microTime(std::chrono::high_resolution_clock::time_point start);
 
 // Argument processing
-napi_status checkArgs(napi_env env, napi_callback_info info, char* methodName,
+napi_status checkArgs(napi_env env, napi_callback_info info, const char* methodName,
   napi_value* args, size_t argc, napi_valuetype* types);
 
 // Async error handling
@@ -79,7 +79,7 @@ struct carrier {
 };
 
 void tidyCarrier(napi_env env, carrier* c);
-int32_t rejectStatus(napi_env env, carrier* c, char* file, int32_t line);
+int32_t rejectStatus(napi_env env, carrier* c, const char* file, int32_t line);
 
 #define REJECT_STATUS if (rejectStatus(env, c, __FILE__, __LINE__) != GRANDIOSE_SUCCESS) return;
 #define REJECT_RETURN if (rejectStatus(env, c, __FILE__, __LINE__) != GRANDIOSE_SUCCESS) return promise;
